pkg/controller/datavolume/smart-clone-controller.go

```diff
@@ -356,9 +356,22 @@ func (r *SmartCloneReconciler) getTargetPVC(dataVolume *cdiv1.DataVolume) (*core
 func newPvcFromSnapshot(dv *cdiv1.DataVolume, name string, snapshot *snapshotv1.VolumeSnapshot, targetPvcSpec *corev1.PersistentVolumeClaimSpec) (*corev1.PersistentVolumeClaim, error) {
 	targetPvcSpecCopy := targetPvcSpec.DeepCopy()
 	restoreSize := snapshot.Status.RestoreSize
-	if restoreSize == nil {
+	if restoreSize == nil || restoreSize.Sign() == -1 {
 		return nil, fmt.Errorf("snapshot has no RestoreSize")
 	}
+	if restoreSize.IsZero() {
+		// Don't error out if the restoreSize is 0. Use the size from the DataVolume instead
+		if dv.Spec.PVC != nil {
+			reqSize := dv.Spec.PVC.Resources.Requests[corev1.ResourceStorage]
+			restoreSize = &reqSize
+		} else if dv.Spec.Storage != nil {
+			reqSize := dv.Spec.Storage.Resources.Requests[corev1.ResourceStorage]
+			restoreSize = &reqSize
+		}
+		if restoreSize.IsZero() {
```

Comment:
I think this is handled in our hooks https://github.com/kubevirt/containerized-data-importer/blob/92382bdffe14b8c106e16724003d1457f84d2d7c/pkg/apiserver/webhooks/datavolume-validate.go#L543-L548